### PR TITLE
recovery email resend and change flow

### PR DESCRIPTION
**Pull request by Hirobreak**
- Make settings request in custom tabs controller and update children
- Display recovery email and status
- resend link confirmation email
- timer to avoid spamming resend
- change recovery email with password popover
- avoid repeated pull down requests

### CODE REVIEW

File: iOS-Email-Client/Controllers/CustomTabsController.swift [hunk 1]
```diff
@@ -11,10 +11,33 @@ import Material
...
+                return
+            }
+            for device in myDevices {
+                let newDevice = Device.fromDictionary(data: device)
+                guard !self.devicesData.devices.contains(where: {$0.id == newDevice.id && $0.active}) else {
+                    continue
+                }
+                self.devicesData.devices.append(newDevice)
+            }
+            self.generalData.recoveryEmail = "pedro.aim93@gmail.com"
```
Comment (GianniCarlo) [inline iOS-Email-Client/Controllers/CustomTabsController.swift]: wat

Comment (Hirobreak) [reply iOS-Email-Client/Controllers/CustomTabsController.swift]: retrieving recoveryEmail is not implemented in the server yet, the same for the recovery email status

File: iOS-Email-Client/Controllers/RecoveryEmailViewController.swift [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    }
+    
+    func showLoader(_ show: Bool){
+        guard show else {
+            resendLoader.stopAnimating()
+            resendLoader.isHidden = true
+            
+            let defaults = UserDefaults.standard
+            let lastTimeResent = defaults.double(forKey: "lastTimeResent")
+            guard lastTimeResent == 0 || Date().timeIntervalSince1970 - lastTimeResent > 300 else {
```
Comment (danieltigse) [inline iOS-Email-Client/Controllers/RecoveryEmailViewController.swift]: magic number

File: iOS-Email-Client/Controllers/RecoveryEmailViewController.swift [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationItem.title = "Recovery Email"
+        navigationItem.leftBarButtonItem = UIBarButtonItem(image: #imageLiteral(resourceName: "arrow-back").tint(with: .white), style: .plain, target: self, action: #selector(goBack))
+        navigationItem.rightBarButtonItem?.setTitleTextAttributes([NSAttributedStringKey.foregroundColor: UIColor.white], for: .normal)
+        recoveryEmailLabel.text = recoveryEmail
+        statusLabel.text = recoveryEmailStatus.description
+        statusLabel.textColor = recoveryEmailStatus.color
+        resendButton.isHidden = recoveryEmailStatus == .verified
+        resendButtonHeightConstraint.constant = recoveryEmailStatus != .pending ? 0.0 : 44.0
```
Comment (danieltigse) [inline iOS-Email-Client/Controllers/RecoveryEmailViewController.swift]: magic number

File: iOS-Email-Client/Controllers/CustomTabsController.swift (hunk 1 above)
Comment (GianniCarlo) [reply iOS-Email-Client/Controllers/CustomTabsController.swift]: I'm not sure how getting your devices relate to your recovery email

File: iOS-Email-Client/Controllers/CustomTabsController.swift [hunk 4]
```diff
@@ -11,10 +11,33 @@ import Material
...
+            self.generalData.recoveryEmailStatus = .pending
```
Comment (GianniCarlo) [inline iOS-Email-Client/Controllers/CustomTabsController.swift]: ^ you even have a pending status, why would you need that placeholder then?

File: iOS-Email-Client/Model/EventData.swift [hunk 5]
```diff
@@ -19,6 +19,7 @@ class EventData {
         var opens = [FeedItem]()
         var modifiedThreadIds = [String]()
         var modifiedEmailKeys = [Int]()
+        var fromWS = false
```
Comment (GianniCarlo) [inline iOS-Email-Client/Model/EventData.swift]: what's this?

Comment (Hirobreak) [reply iOS-Email-Client/Model/EventData.swift]: if the event response comes from the websocket, i do not ack the event nor update the loading flag in mailbox

File: iOS-Email-Client/Controllers/CustomTabsController.swift (hunk 1 above)
Comment (Hirobreak) [reply iOS-Email-Client/Controllers/CustomTabsController.swift]: getDevices will soon change into something like getSettings, you could wait for this change in the server but lot of prs are gonna stacks

**Comment by danieltigse** [comment]
Don't merge this PR until `/user/settings` is implemented

**Pull request merged**